quiz/backend/entrypoint.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+if [ "$DATABASE" = "postgres" ]
+then
+  echo "Waiting for postgres ..."
+
+  while ! nc -z $POSTGRES_HOST $POSTGRES_PORT; do
+    sleep 0.1
+    done
+
+    echo "Postgres started"
+
+fi
+python manage.py collectstatic --no-input --clear
+python manage.py flush --no-input
```

Comment:
wydaje mi się że dump bazy (chyba to flush robił) i auto migracja w wersji produkcyjnej jak i devowej może być nieco problemem.
Jakoś tak nie chciałbym żeby mi się baza na produkcji usunęła jak uruchomie obraz.

Reply:
Wydaje się, że zgodnie z logiką, jak zmieni się kod do bazy, a baza nie zmigruje, to się sypnie aplikacja, więc automatycznie ma sens. Natomiast jak się odpali automatycznie i nikt tego nie przetestuje, a będzie to niezgodne z intencją autora (np usunie kolumne) to dane się same skasują, więc wydaje się, że automatyczne są ryzykowne.

Jaka zatem decyzja? Idziemy w CI/CD, nie będzie człowieka, który to robi, co wyklucza ręczną migrację. Moim zdaniem automat, który przed deployem odpala testy, czy wszystko jest w porządku. Poprawcie mnie, jeżeli nie wziąłem czegoś pod uwagę?

Reply:
Flusha chętnie wywalę, bo sama nie byłam do tego przekonana, ale automatyczne migracje z wcześniejszymi testami myślę, że mają sens.